doc/design/software-bridge-management.md

```diff
@@ -0,0 +1,460 @@
+---
+title: software bridge management
+authors:
+  - ykulazhenkov
+reviewers:
+creation-date: 15-02-2024
+last-updated: 15-02-2024
+---
+
+# Software bridge management
+
+## Summary
+
+When NIC is configured to switchdev mode, a VF representor net device is created for each VF on it. 
+These representors are used by a software switch (OVS, Linux bridge) to control traffic and configure hardware offloads. 
+The software bridge is an essential part of using NICs in switchdev mode.
+
+**sriov-network-operator** can set switchdev mode for a NIC and create VFs on it, 
+but it doesn't provide any functionality to create and configure software bridges. 
+
+This document contains a proposal to add limited support for software bridges configuration to the **sriov-network-operator**.
+
+This feature assumes integration with [ovs-cni](https://github.com/k8snetworkplumbingwg/ovs-cni) and 
+[accelerated-bridge-cni](https://github.com/k8snetworkplumbingwg/accelerated-bridge-cni).
+
+Depends on [_switchdev and systemd modes refactoring_](switchdev-refactoring.md) feature.
+
+## Motivation
+
+SRIOV Legacy mode is no longer actively developed, and we need to encourage users to migrate to switchdev mode,
+which is actively developed and will continue to receive new features and improvements.
+
+To promote switching to switchdev configurations, we need to provide a nice UX for the end user.
+This requires providing an easy way to configure software switches, which are prerequisites for NICs in switchdev mode.
+
+
+### Use Cases
+
+* As a user, I expect that **sriov-network-operator** will install `ovs-cni` and `accelerated-bridge-cni` to hosts.
+* As a user, I want to create `OVSNetwork` CR, which will result in creation of `NetworkAttachmentDefinition` CR that uses
+`ovs-cni` and contains required resource request.
+* As a user, I want to create `BridgeNetwork` CR which will result in creation of `NetworkAttachmentDefinition` CR that uses
+`accelerated-bridge-cni` and contains the required resource requests.
+* As a user, I want to define configuration for software bridges inside the `SriovNetworkNodePolicy` CR and expect that the
+operator will create required bridges, configure them, and attach uplinks (physical functions).
+
```

Comment:
What about the OVN-K CNI use case? Will the api to set it remain the same?
Will OVN-K and OVS-CNI have any conflicts if OVN-K is installed in the first place.

Reply:
I believe that the only difference for OVN-k use case is that sriov-network-operator will need to be configured to `systemd` mode (check [this doc](https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/doc/design/switchdev-refactoring.md) for details).  [NIC configuration only flow](https://github.com/ykulazhenkov/sriov-network-operator/blob/3753c5dcc4cf83a46a85bf1334f6ada31d9a22ce/doc/design/software-bridge-management.md?plain=1#L110) is proposed to be compatible with existing OVN-k use-case. In this case NIC will be configured to switchdev mode early on boot by the systemd service, but the operator will not manage software bridges.

Reply:
The current proposal assumes that "Fully automatic" flow can be used only for dedicated NICs (not used by OVN-k) which are fully controlled by the sriov-network-operator.